crates/sui-adapter/src/execution_engine.rs

```diff
@@ -574,18 +560,23 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
         temporary_store.drop_writes();
         // Must reset the storage rebate since we are re-executing.
         gas_status.reset_storage_cost_and_rebate();
-        let advance_epoch_safe_mode_pt =
-            construct_advance_epoch_safe_mode_pt(&params, protocol_config)?;
-        programmable_transactions::execution::execute::<_, execution_mode::System>(
-            protocol_config,
-            move_vm,
-            temporary_store,
-            tx_ctx,
-            gas_status,
-            None,
-            advance_epoch_safe_mode_pt,
-        )
-        .expect("Advance epoch with safe mode must succeed");
+
+        if protocol_config.advance_epoch_safe_mode_in_rust() {
+            temporary_store.advance_epoch_safe_mode(&params);
```

Comment:
This PR removes the advance_epoch_safe_mode from Move code, so yes I believe we must gate on the protocol version because an older version will attempt to call Move

Reply:
I think we would need a protocol version bump if you were adding a move implementation (to wait for the protocol version bump that introduced the new framework version), but I think if you are removing it, it should be okay, as long as the Rust implementation behaves like the move one.

Reply:
hmm I see. I guess we could reuse the check on version 2 where we fixed the epoch start timestamp.
My only concern is then whether I have a bug such that the two impls are different somehow